mergin/cli.py

```diff
@@ -51,132 +68,183 @@ def pretty_diff(diff):
+            fg="red",
+        )
         return None
-
     return mc
 
 
 def _print_unhandled_exception():
     """ Outputs details of an unhandled exception that is being handled right now """
-    click.secho("Unhandled exception!", fg='red')
+    click.secho("Unhandled exception!", fg="red")
     for line in traceback.format_exception(*sys.exc_info()):
         click.echo(line)
 
 
 @click.group()
-def cli():
-    pass
-
-
-@cli.command()
-@click.argument('url', required=False)
-@click.option('--login', prompt=True)
-@click.option('--password', prompt=True, hide_input=True)
-def login(url, login, password):
-    """Fetch new authentication token. If no URL is specified, the public Mergin instance will be used."""
-    c = MerginClient(url)
-    click.echo("Mergin URL: " + c.url)
-    if not c.is_server_compatible():
-        click.secho(str('This client version is incompatible with server, try to upgrade'), fg='red')
-        return
-    try:
-        session = c.login(login, password)
-    except LoginError as e:
-        click.secho('Unable to log in: ' + str(e), fg='red')
-        return
-
-    print('export MERGIN_URL="%s"' % c.url)
-    print('export MERGIN_AUTH="%s"' % session['token'])
+@click.option(
+    "--url",
+    envvar="MERGIN_URL",
+    default="https://public.cloudmergin.com",
+    help="Mergin server URL. Default is: https://public.cloudmergin.com",
+)
+@click.option("--auth_token", envvar="MERGIN_AUTH", help="Mergin authentication token string")
+@click.option("--username", envvar="MERGIN_USERNAME")
+@click.option("--password", cls=OptionPasswordIfUser, prompt=True, hide_input=True, envvar="MERGIN_PASSWORD")
+@click.option(
+    "--show_token",
+    is_flag=True,
+    help="Flag for echoing the authentication token. Useful for setting the MERGIN_AUTH environment variable.",
+)
+@click.pass_context
+def cli(ctx, url, auth_token, username, password, show_token):
+    """
+    Command line interface for the Mergin client module.
+    For user authentication on server, username and password need to be given either as environment variables
+    (MERGIN_USERNAME, MERGIN_PASSWORD), or as options (--username, --password).
+    To set MERGIN_AUTH variable, run the command with --show_token option to see the token and how to set it manually.
+    """
+    mc = get_client(url=url, auth_token=auth_token, username=username, password=password, show_token=show_token)
+    ctx.obj = {"client": mc}
 
 
 @cli.command()
-@click.argument('project')
-@click.option('--public', is_flag=True, default=False, help='Public project, visible to everyone')
-def create(project, public):
-    """Create a new project on Mergin server"""
-
-    c = _init_client()
-    if c is None:
+@click.argument("project")
+@click.option("--public", is_flag=True, default=False, help="Public project, visible to everyone")
+@click.option("--namespace", help="Namespace for the new project. Default is current Mergin user namespace.")
```

Comment:
I would skip this option, so that there is one preferred way how to refer to project names (using slash notation) - otherwise we should support `--namespace` in other commands too